lib/astronoby/angle.rb

```diff
@@ -4,57 +4,97 @@
 
 module Astronoby
   class Angle
-    UNITS = [
-      DEGREES = :degrees,
-      HOURS = :hours,
-      RADIANS = :radians
-    ].freeze
-
-    UNIT_CLASS_NAMES = {
-      DEGREES => "Astronoby::Degree",
-      HOURS => "Astronoby::Hour",
-      RADIANS => "Astronoby::Radian"
-    }
-
     PRECISION = 14
-    PI = BigMath.PI(10)
+    PI = BigMath.PI(PRECISION)
+    FORMATS = %i[dms hms].freeze
 
     class << self
-      UNIT_CLASS_NAMES.each do |unit, class_name|
-        define_method("as_#{unit}") do |angle|
-          Kernel.const_get(class_name).new(angle)
-        end
+      def as_radians(radians)
+        new(radians)
+      end
+
+      def as_degrees(degrees)
+        radians = degrees / BigDecimal("180") * PI
```

Comment:
Thoughts on having a constant like `DEGREES_PER_RADIAN = BigDecimal("180") * PI`? (I'm not sure I used the correct name)

Reply:
I chose to add a `PI_IN_DEGREES = BigDecimal("180")` constant because I wasn't able to add something with `PI` as its use is different depending on the conversion direction (multiply or divide).